### PR TITLE
Add error handling for overflows

### LINKED ISSUE #189: [PE-D][Tester E] Balance is not updating when it becomes too big
The balance stops updating and no error is thrown when a certain number limit is hit.

<i><video controls><source src="https://raw.githubusercontent.com/rubinnn/ped/main/files/b25cf7b3-c2a6-49b0-8b06-3e4b69fc110b.mov" type="video/mp4">Your browser does not support the video tag.</video><br>video:https://raw.githubusercontent.com/rubinnn/ped/main/files/b25cf7b3-c2a6-49b0-8b06-3e4b69fc110b.mov</i>

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: rubinnn/ped#1

### LINKED ISSUE #220: [PE-D][Tester E] Summary is unable to calculate due to the large number
The summary is unable to calculate due to the large number inputted. The summary just rounds to the largest number available

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: rubinnn/ped#8

### PR DESCRIPTION
**Pull request by beginner-web-developer**
Fixes #189 

We consider this as a bug, as it may be possible for the user to hit a very large amount for the balance, even if individual transaction amounts are small and not extreme.

Fixes #220 
`summary` is the sum of a subset of transactions, so it is sufficient to check the overflow for balance.

### CODE REVIEW

**Comment by reginateh** [comment]
LGTM

**Comment by reginateh** [comment]
If we have this check then don't need to handle overflow in `summary` right? Because `summary` is the sum of a subset of transactions.

**Comment by beginner-web-developer** [comment]
i think so

File: src/main/java/seedu/address/model/client/Client.java [hunk 1]
```diff
@@ -110,6 +110,17 @@ public void updateBalance(double amount) {
...
     }
 
+    /**
+     * Checks if the transaction causes a double overflow.
+     *
+     * @param amount The amount of the transaction to be added / deleted.
+     * @return whether the balance would overflow.
+     */
+    public boolean checkIsOverflow(double amount) {
+        double updatedBalance = balance + amount;
```
Comment (reginateh) [inline src/main/java/seedu/address/model/client/Client.java:120]: Should we first check if the `amount` overflows? If `amount` is `INFINITY` it is not safe to add it with balance

Comment (beginner-web-developer) [reply src/main/java/seedu/address/model/client/Client.java:120]: but adding infinity with anything is still infinity i think?

Comment (reginateh) [reply src/main/java/seedu/address/model/client/Client.java:120]: `positiveInfinity` + `negativeInfinity` is `NaN`
I know the balance is not infinity but just for safety?

**Comment by reginateh** [comment]
Fixes #220

File: src/main/java/seedu/address/model/client/Client.java (hunk 1 above)
Comment (beginner-web-developer) [reply src/main/java/seedu/address/model/client/Client.java:123]: ok I added the check

**Comment by reginateh** [comment]
LGTM

**Pull request merged**